### PR TITLE
Fat 1496 Import a large EDIFACT invoice file

### PR DESCRIPTION
**Pull request by TetianaParanich**

### CODE REVIEW

**Review by Javokhir12 — APPROVED** [review]

File: cypress/support/fragments/data_import/mapping_profiles/fieldMappingProfiles.js [hunk 1]
```diff
@@ -97,6 +97,22 @@ export default {
     closeViewModeForMappingProfile();
     cy.expect(actionsButton.exists());
   },
+
+  createLargeInvoiceMappingProfile:(mappingProfileName) => {
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/mapping_profiles/fieldMappingProfiles.js:101]: Done

[the remark this answers is not in the document]
Comment (dmitriyfolio) [reply cypress/support/fragments/data_import/mapping_profiles/fieldMappingProfiles.js:101]: can't see changes

File: cypress/support/fragments/data_import/logs.js [hunk 2]
```diff
@@ -83,6 +83,10 @@ export default {
     });
   },
 
+  checkQuantityRecordsInFile:() => {
+    cy.do(MultiColumnListCell({ row: 0, column: '18' }).exists());
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs.js:87]: Done

File: cypress/integration/data-import/mrc-import-with-matching-001.spec.js [hunk 3]
```diff
@@ -16,6 +15,8 @@ import exportFile from '../../support/fragments/data-export/exportFile';
 import ExportMarcFile from '../../support/fragments/data-export/export-marc-file';
 import FileManager from '../../support/utils/fileManager';
 import testTypes from '../../support/dictionary/testTypes';
+import settingsMenu from '../../support/fragments/settingsMenu';
```
Comment (dmitriyfolio) [inline cypress/integration/data-import/mrc-import-with-matching-001.spec.js:18]: use capital or minuscule first letter in all cases as template

File: cypress/integration/data-import/mrc-import-with-matching-999ff.spec.js [hunk 4]
```diff
@@ -47,7 +48,7 @@ describe('ui-data-import: MARC file import with matching for 999 ff field', () =
       name: mappingProfileNameForExport,
       typeValue : NewFieldMappingProfile.folioRecordTypeValue.instance,
     };
-    SettingsDataImport.goToMappingProfiles();
+    cy.visit(`${settingsMenu.mappingProfilePath}`);
```
Comment (dmitriyfolio) [inline cypress/integration/data-import/mrc-import-with-matching-999ff.spec.js:51]: you don't need to use string template in such cases. You can write string const directly.

File: cypress/support/api/invoice.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+Cypress.Commands.add('getInvoiceIdApi', (searchParams) => {
+  return cy
+    .okapiRequest({
+      path: 'invoice/invoices',
+      searchParams,
+    })
+    .then(responce => {
+      cy.log(JSON.stringify(responce));
```
Comment (dmitriyfolio) [inline cypress/support/api/invoice.js:8]: remove logging and pause from the result code

Comment (TetianaParanich) [reply cypress/support/api/invoice.js:8]: Done

File: cypress/integration/data-import/mrc-import-with-matching-001.spec.js (hunk 3 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-with-matching-001.spec.js:18]: Done

File: cypress/integration/data-import/mrc-import-with-matching-999ff.spec.js (hunk 4 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/mrc-import-with-matching-999ff.spec.js:51]: Done

File: cypress/support/fragments/data_import/logs.js [hunk 6]
```diff
@@ -48,6 +48,10 @@ const checkIsInvoiceCreated = () => {
   cy.expect(MultiColumnListCell({ row: 0, column: 'Created' }).exists());
 };
 
+const quantityRecordsInInvoice = {
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/support/fragments/data_import/logs.js]: Thanks!

File: cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js [hunk 7]
```diff
@@ -24,7 +24,15 @@ describe('ui-data-import: EDIFACT file import with creating of new invoice recor
     );
   });
 
-  it('C345423 EDIFACT file import with creating of new invoice record', { tags: [testTypes.smoke] }, () => {
+  afterEach(() => {
+    const invoiceNumberFromEdifactFile = '94999';
+    cy.getInvoiceApi({ query: `vendorInvoiceNo="${invoiceNumberFromEdifactFile}"` })
+      .then(({ body }) => {
```
[the remark this answers is not in the document]
Comment (TetianaParanich) [reply cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js]: Done

File: cypress/integration/data-import/edi-import-large-invoice-file.spec.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    FieldMappingProfiles.deleteFieldMappingProfile(mappingProfileName);
+  });
+
+  it('C347615 Import a large EDIFACT invoice file', { tags: [TestTypes.smoke] }, () => {
+    // unique file name to upload
+    const fileName = `autotestFile.${getRandomPostfix()}.edi`;
+
+    // create Field mapping profile
+    cy.visit(SettingsMenu.mappingProfilePath);
+    FieldMappingProfiles.createInvoiceMappingProfile(mappingProfileName, FieldMappingProfiles.mappingProfileForDuplicate.harrassowitz, newMappingProfile.organization.harrassowitz);
```
Comment (dmitriyfolio) [inline cypress/integration/data-import/edi-import-large-invoice-file.spec.js:46]: fix issues from Linter

File: cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js [hunk 9]
```diff
@@ -34,50 +40,50 @@ describe('ui-data-import: EDIFACT file import with creating of new invoice recor
     const fileName = `C345423autotestFile.${getRandomPostfix()}.edi`;
 
     // create Field mapping profile
-    settingsDataImport.goToMappingProfiles();
-    fieldMappingProfiles.waitLoading();
-    fieldMappingProfiles.createInvoiceMappingProfile(mappingProfileName);
-    fieldMappingProfiles.checkMappingProfilePresented(mappingProfileName);
+    cy.visit(SettingsMenu.mappingProfilePath);
+    FieldMappingProfiles.waitLoading();
+    FieldMappingProfiles.createInvoiceMappingProfile(mappingProfileName, FieldMappingProfiles.mappingProfileForDuplicate.gobi, newMappingProfile.organization.gobiLibrary);
```
Comment (dmitriyfolio) [inline cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js:45]: fix issue from Linter

File: cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js [hunk 10]
```diff
@@ -24,7 +25,13 @@ describe('ui-data-import: EDIFACT file import with creating of new invoice recor
     );
   });
 
-  it('C345423 EDIFACT file import with creating of new invoice record', { tags: [testTypes.smoke] }, () => {
+  afterEach(() => {
+    const invoiceNumberFromEdifactFile = '94999';
```
Comment (dmitriyfolio) [inline cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js:29]: may be this constant should locate somewhere in fragments

**Review by dmitriyfolio — APPROVED** [review]

File: cypress/support/api/invoice.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+Cypress.Commands.add('getInvoiceIdApi', (searchParams) => {
+  return cy
+    .okapiRequest({
+      path: 'invoice/invoices',
+      searchParams,
+    })
+    .then(responce => {
+      cy.pause();
```
Comment (dmitriyfolio) [inline cypress/support/api/invoice.js:8]: excess code

File: cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js (hunk 10 above)
Comment (TetianaParanich) [reply cypress/integration/data-import/edi-import-file-for-create-invoice-record.spec.js]: Done

**Pull request merged**